fission/src/mirabuf/MirabufLoader.ts

```diff
@@ -27,20 +34,43 @@ type MapCache = { [id: MirabufCacheID]: MirabufCacheInfo }
 
 const robotsDirName = "Robots"
 const fieldsDirName = "Fields"
+const piecesDirName = "Pieces"
 const root = await navigator.storage.getDirectory()
-const robotFolderHandle = await root.getDirectoryHandle(robotsDirName, { create: true })
-const fieldFolderHandle = await root.getDirectoryHandle(fieldsDirName, { create: true })
-
-export let backUpRobots: MapCache = {}
-export let backUpFields: MapCache = {}
+const robotFolderHandle = await root.getDirectoryHandle(robotsDirName, {
+    create: true,
+})
+const fieldFolderHandle = await root.getDirectoryHandle(fieldsDirName, {
+    create: true,
+})
+const pieceFolderHandle = await root.getDirectoryHandle(piecesDirName, {
+    create: true,
+})
+
+const dirNameMap = new Map([
+    [MiraType.ROBOT, robotsDirName],
+    [MiraType.FIELD, fieldsDirName],
+    [MiraType.PIECE, piecesDirName],
+])
+
+export const backUpRobots: MapCache = {}
+export const backUpFields: MapCache = {}
+export const backUpPieces: MapCache = {}
+
+export const backUpMap = new Map([
+    [MiraType.ROBOT, backUpRobots],
+    [MiraType.FIELD, backUpFields],
+    [MiraType.PIECE, backUpPieces],
+])
 
 export const canOPFS = await (async () => {
     try {
         if (robotFolderHandle.name == robotsDirName) {
             robotFolderHandle.entries
             robotFolderHandle.keys
```

Comment:
so you didn't do this but what is going on here?

Reply:
Great question to Ayla, I should probably open a ticket to refactor the caching service